framework_tvb/tvb/tests/framework/core/services/operation_service_test.py

```diff
@@ -83,33 +85,40 @@ def _assert_stored_ddti(self, expected_cnt=1):
         assert datatype.subject == DataTypeMetaData.DEFAULT_SUBJECT, "Wrong data stored."
         return datatype
 
-    def test_datatypes_groups(self, test_adapter_factory):
+    def test_datatypes_groups(self, test_adapter_factory, datatype_group_factory):
         """
         Tests if the dataType group is set correct on the dataTypes resulted from the same operation group.
         """
         all_operations = dao.get_filtered_operations(self.test_project.id, None)
         assert len(all_operations) == 0, "There should be no operation"
 
-        algo = test_adapter_factory(TestAdapter3)
-        adapter_instance = ABCAdapter.build_adapter(algo)
-        view_model = TestModel()
-        data = {model_burst.RANGE_PARAMETER_1: 'param_5', 'param_5': [1, 2]}
-        # Create Group of operations
-        OperationService().fire_operation(adapter_instance, self.test_user, self.test_project.id,
-                                          view_model=view_model, **data)
+        dt_group = datatype_group_factory(project=self.test_project)
+        model = TestModel()
+        test_adapter_factory()
+        adapter = TestFactory.create_adapter("tvb.tests.framework.adapters.testadapter1", "TestAdapter1")
```

Comment:
Isn't `datatype_group_factory` generating view_models and using existing algorithms already?